packages/core/src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -73,13 +77,25 @@ export default class GooglePayPaymentProcessor {
         });
     }
 
-    updatePaymentDataRequest(payloadToUpdate: { currencyCode: string; totalPrice: string }) {
-        const paymentDataRequest = this._getPaymentDataRequest();
+    updatePaymentDataRequest(paymentDataRequest: Partial<GooglePayPaymentDataRequestV2>) {
```

Comment:
I believe here we can try using a union type and validate payload using a typeguard in the method.

Reply:
There are a lot of fields to pass if we remove partial. We use this method just to update some parts of class variable only.

Reply:
It is not a big deal if it will be empty object, because this method are used to update `existingPaymentDataRequest`. 

In this case `this._paymentDataRequest` will be equal to `existingPaymentDataRequest` that is good as well.

The main point, why we add <Partial> here, because we should have an ability to pass only the data, that we want to be updated. But if we make it as union, we should use typeguards all over the place to let Typescript know that we pass "full" `paymentDataRequest` data, what is wrong in my opinion.

What do you think?